src/rule-generated-flow-types.js

```diff
@@ -442,7 +442,9 @@ module.exports = {
       /**
        * Find useQuery() calls without type arguments.
        */
-      'CallExpression[callee.name=useQuery]:not([typeArguments])'(node) {
+      'CallExpression[callee.name=useQuery]:not(:has([typeParameters], [typeArguments]))'(
```

Comment:
After spending some time on #115, there are edge cases not caught in the way this PR is currently done:
The Flow AST and typescript AST have differences in how they name certain nodes (`ObjectTypeReference` vs `TSTypeLiteral` eg). And even beyond that, they have different structures in the nodes at some parts (see #115, especially the function `validateObjectTypeAnnotation` in both rule variants).

There might be a way to carefully integrate the differences into one rule. But I want to let that decision  up to the maintainers to decide, what they prefer more. If the one-rule way is chosen, I propose to copy the test from #115, so more cases are checked here.
(That is how I found all the small differences, the test there are basically the complete test suite for `generated-flow-types` from `main`, changed so it works with TS).

Cheers Hannes